lldb/source/Plugins/ExpressionParser/Clang/ClangUserExpression.cpp

```diff
@@ -553,9 +553,9 @@ bool ClangUserExpression::PrepareForParsing(
 }
 
 bool ClangUserExpression::TryParse(
-    DiagnosticManager &diagnostic_manager, ExecutionContextScope *exe_scope,
-    ExecutionContext &exe_ctx, lldb_private::ExecutionPolicy execution_policy,
-    bool keep_result_in_memory, bool generate_debug_info) {
+    DiagnosticManager &diagnostic_manager, ExecutionContext &exe_ctx,
+    lldb_private::ExecutionPolicy execution_policy, bool keep_result_in_memory,
+    bool generate_debug_info) {
```

Comment:
Jim made comments below. `ExecutionContextScope *` is just as good as an ExecutionContext, but it is actually more specific as a `Target`, `Process`, `Thread` and `StackFrame` are all `ExecutionContextScope *` objects. They just know how to re-create any items needed. So a `StackFrame` can use `ExecutionContextScope` APIs (which is virtually overrides) to get the thread, process and target. The `Thread` can get the process and target. The `Process` can get the target. So it is no better or worse, it just specifies one thing would be the same as having an `ExecutionContext` filled out as much as needed. We do need to be careful to pass in the right thing as Jim said, where if we only want the `Target` to be used, we shouldn't pass in a `StackFrame` as the `ExecutionContextScope *`, but this will be exactly the same as passing a `ExecutionContext` that has the stack frame filled out when we only want to consult the target. So no real difference, and using `ExecutionContextScope *` is better since we can control what we pass as the root of where to start looking whereas if we pass in a fully filled out ExecutionContext, we will always consult the deepest item (Stack, then frame, then thread, then process/target) if it is filled out. 

So if the ExecutionContext is always filled out correctly to the right depth, then calling the `exe_ctx.GetBestExecutionContextScope()` suggestion will work. If not, then we need to pass in the right level, or modify the ExecutionContext and remove items from it.

Reply:
I removed this parameter because it seemed redundant as it was only used for the `ClangExpressionParser`, which wants access to all of Process/Target/StackFrame. So if we can get that from the ExecutionContext, there's no point in passing the `exe_scope` down the the parser. Maybe I'm misunderstanding but my impression was that this is a simplification and wouldn't cause any functional change. This is purely about making the `ClangExpressionParser` get the `StackFrame` calculation right, which it doesn't even currently use apart for logging

Reply:
I was also mostly moved by the fact that there were two things specifying what should be one thing.  I was hoping that this was just redundant, not someone being overly clever, and Michael showed that it wasn't a used distinction, which is good.  As far as which type to use, that's really more convenience, right?  If the methods you want to call immediately from TryParse take ExecutionContextScopes, use that, if ExecutionContext's, use that.  I don't think it makes any other difference.